toolchains/workspace-pnpm/macros.bzl

```diff
@@ -581,6 +581,45 @@ def _npm_test_impl(
         cmd_args([build_context.workspace_root, ctx.label.package], delimiter = "/"),
         "--bin",
         cmd_args(program_run_info),
+        "--",
+        program_args,
+    ])
+
+    args_file = ctx.actions.write("args.txt", run_cmd_args)
+
+    return inject_test_run_info(
+        ctx,
+        ExternalRunnerTestInfo(
+            type = test_info_type,
+            command = [run_cmd_args],
+        ),
+    ) + [
+        DefaultInfo(default_output = args_file),
+    ]
+
+def _npm_build_and_test_impl(
```

Comment:
I tried rolling this into the `_npm_test_impl` and ran into 2 issues:
- I had to add the `runnable_tsc_build ` attr to 3 rules that didn't need it
- Even after I added it, there was no way I could find to set a default falsy value for that attr to switch on inside `_npm_test_impl`

Also, the conditional branching was visually significant enough inside the function that I think it makes sense to leave these as separate functions in this instance.